src/base/PhpBlockView.php

```diff
@@ -254,9 +264,63 @@ public function extraValue($key, $defaultValue = null, $template = false)
      * ```
      *
      * @return \luya\web\View The global application View Object which is also the same as the layout or cmslayout.
+     * @deprecated Please use `MyBlockAsset::register($this);` instead.
+     * @since 1.0.5 Deprecated.
      */
     public function getAppView()
     {
-        return Yii::$app->getView();
+        return $this;
+    }
+
+    /**
+     * @return array
+     * @since 1.0.5
+     */
+    public function getBlockAssets()
+    {
+        return [
+            'metaTags' => $this->metaTags,
+            'linkTags' => $this->linkTags,
+            'cssFiles' => $this->cssFiles,
+            'css' => $this->css,
+            'jsFiles' => $this->jsFiles,
+            'js' => $this->js,
+        ];
+    }
+
+    /**
+     * @param array $blockAssets
+     * @param array $assetBundles
+     * @throws \yii\base\InvalidConfigException
+     * @since 1.0.5
+     */
+    public static function registerToAppView(array $blockAssets, array $assetBundles)
+    {
+        $appView = Yii::$app->view;
+
+        foreach ($blockAssets as $attribute => $blockAsset) {
+            if (!empty($blockAsset)) {
+                
+                if ($attribute == 'js' || $attribute == 'jsFiles') {
+                    /**
+                     * js and jsFiles must keep the array keys as position and have subarray
+                     * @see \yii\web\View::POS_HEAD
+                     */
+
+                    $appAssets = &$appView->{$attribute};
```

Comment:
Because in line 313 I write back into the `$appAssets` array. Otherwise I have to use `$appView->{$attribute}[$key]` and this is harder to read I mean.